_posts/en/newsletters/2019-07-31-newsletter.md

```diff
@@ -0,0 +1,246 @@
+  where we lamented the lack of a BIP322 implementation.
+
+    Based on early feedback on the PR, BIP322 has also been updated so
+    that signing and verifying messages with legacy P2PKH keys uses the
+    old signmessage format.  This allows BIP322 tools to be fully backwards
+    compatible with the existing and widely-implemented signmessage standard
+    (which only supports P2PKH).
+
+- **Bloom filter discussion:** in the *notable changes* section of [last
+  week's newsletter][Newsletter #56], we mentioned a [merged PR][Bitcoin
+  Core #16152] that disabled bloom filters in Bitcoin Core's default
+  configuration.  The author of the PR [announced][bloom announce] this
+  unreleased change to the Bitcoin-Dev mailing list and several people
+  replied with questions or concerns.  Some takeaways from the
+  discussion include:
+
+    - *No urgent action required:* as of this writing, over 20% of nodes
+      accepting incoming connections (listening nodes) are running a
+      version of Bitcoin Core that's over a year old [according to
+      BitNodes][bitnodes dashboard].  At least 5% are running a version
+      that's over two years old.  Extrapolating to the future, that
+      means there will still be over 500 listening nodes providing
+      bloom filters at the beginning of 2022.  So, even with no further
+      action, wallet developers may have a significant amount of time in
+      which to adapt their programs.
+
+    - *Spies likely to run their own nodes:* additional to non-upgraded
+      nodes, it's possible that block chain analysis companies will
+      continue to run their own nodes providing bloom filter support in
+      the future so that they can collect statistics from wallets whose
+      filters [leak information][filter privacy] about what addresses
+      they contain.
+
+    - *DNS seeds can return only nodes signaling BIP111:* most P2P light
+      wallets query one or more Bitcoin DNS seeds for a list of nodes to
+      use.  Some seeds allow filtering which nodes they return by what
+      service bits that node has configured[^dns-query], with the
+      service bit for bloom filter support being specified by
+      [BIP111][].
+
+    - *BIP157 would use more bandwidth than BIP37:* it was suggested
+      that some users of [BIP37][] bloom filters will be unable to
+      switch to [BIP157][] compact block filters because of the later's
+      higher bandwidth usage.  Optech briefly investigated filter size
+      in [Newsletter #43][] and found the filters themselves to use less
+      than 100 MB a month.  Clients would also need to download matching
+      blocks, which would vary depending on their activity (roughly one
+      block for every sent or received transaction for typical users; most
+      blocks are less than 2 MB in size).  It's unclear how many users
+      would be significantly affected by having to download that amount
```

Comment:
relatively modest amount of data? (seems quite light)

Reply:
Some people on the list were arguing that it's heavy in less developed parts of the world.  I think we want to deliver the facts here without an explicit judgment call so that readers can come to their own conclusions.

Reply:
Gotcha. I did indeed have to think more :) but makes sense.